### DISCUSSION TITLE
link trace without stack problem

### DISCUSSION THREAD

**Discussion by luibgn** [bochs-emu/Bochs#370] [Ideas]
Why not using

#define BX_LINK_TRACE(i) {                             \
  BX_COMMIT_INSTRUCTION(i);                            \
  return linkTrace(i);                                 \
}

same as normal instruction chaining? That avoids the need of the depth limits.
I'm trying it and it seems working well. Also with MP.
Am I missing something?

**Comment by stlintel** [comment]
This reduce the problem but not solve it.
I've seen many issues reported due to excessive trace linking.
The most severe was related to floating point instruction handlers which have local variables, sometimes a lot of local variables.
Linking many such instructions will lead to stack overflow always, doesn't matter how BX_LINK_TRACE is done.

BTW, it would require linkTrace() method to return smth, right ?

**Comment by luibgn** [comment]
return void is just fine. It is like the normal chaining that you are doing for years.

#define BX_EXECUTE_INSTRUCTION(i) {                    \
  BX_INSTR_BEFORE_EXECUTION(BX_CPU_ID, (i));           \
  RIP += (i)->ilen();                                  \
  return BX_CPU_CALL_METHOD(i->execute1, (i));         \
}

If you still see a stack problem with the "return linkTrace(i);" I think that there is a mistake in some point in the chaining and that should be investigates and solved.
If all chaining is ok I think that you always should have only 1 depth in the stack no matter what how many pseudo-recursive calls you do.

I din't know this chaining trick. You told me a few years ago in an old conversation.
